src/cmap/connection.ts

```diff
@@ -645,7 +645,7 @@ function supportsOpMsg(conn: Connection) {
     return false;
   }
 
-  return maxWireVersion(conn) >= 6 && !description.__nodejs_mock_server__;
+  return !description.__nodejs_mock_server__;
```

Comment:
```suggestion
  return maxWireVersion(conn) >= 6 && !description.__nodejs_mock_server__;
```
We actually need to leave this check in place.  For the initial connection handshake, we need to use OP_QUERY instead of OP_MESSAGE